consensus/spos/bls/subroundSignature_test.go

```diff
@@ -668,6 +679,299 @@ func TestSubroundSignature_DoSignatureJobWithMultikey(t *testing.T) {
 	})
 }
 
+func TestSubroundSignature_SendSignatureShouldErr(t *testing.T) {
+
+	container := mock.InitConsensusCore()
+	expErr := fmt.Errorf("Error")
+	container.SetSigningHandler(&consensusMocks.SigningHandlerStub{
+		CreateSignatureShareForPublicKeyCalled: func(message []byte, index uint16, epoch uint32, publicKeyBytes []byte) ([]byte, error) {
+			return make([]byte, 0), expErr
+		},
+	})
+	consensusState := initConsensusStateWithKeysHandler(
+		&testscommon.KeysHandlerStub{
+			IsKeyManagedByCurrentNodeCalled: func(pkBytes []byte) bool {
+				return true
+			},
+		},
+	)
+
+	ch := make(chan bool, 1)
+
+	sr, _ := spos.NewSubround(
+		bls.SrBlock,
+		bls.SrSignature,
+		bls.SrEndRound,
+		int64(70*roundTimeDuration/100),
+		int64(85*roundTimeDuration/100),
+		"(SIGNATURE)",
+		consensusState,
+		ch,
+		executeStoredMessages,
+		container,
+		chainID,
+		currentPid,
+		&statusHandler.AppStatusHandlerStub{},
+	)
+	sr.Header = &block.Header{}
+
+	signatureSentForPks := make(map[string]struct{})
+	srSignature, _ := bls.NewSubroundSignature(
+		sr,
+		&statusHandler.AppStatusHandlerStub{},
+		&testscommon.SentSignatureTrackerStub{
+			SignatureSentCalled: func(pkBytes []byte) {
+				signatureSentForPks[string(pkBytes)] = struct{}{}
+			},
+		},
+		&mock.SposWorkerMock{},
+	)
+
+	mockNumMultikey := int32(0)
+
+	err := srSignature.SendSignature(0, "a", &mockNumMultikey)
+
+	assert.Equal(t, expErr, err)
+}
+
+func TestSubroundSignature_SendSignatureShouldErrorCreateAndSendSignMessage(t *testing.T) {
```

Comment:
add `t.Parallel()` for tests?

Reply:
updated

Reply:
please add `t.Parallel()` to the other tests as well